### ISSUE TITLE
I can't install Pycharm ??

### ISSUE THREAD

**Issue by fabianmendes** [Botspot/pi-apps#480]
It says  _permission denied_ like as it would need `sudo`, Idk what'd happen

![2021-03-18-223230_800x480_scrot](https://user-images.githubusercontent.com/31230743/111778399-f871bd00-888a-11eb-9bf5-5cfeb684cf8f.png)

**Comment by Itai-Nelken** [comment]
weird! @Botspot  fixed that in commit 8c2034cfc9d196bc51496906b46421f1995bcc61.
can you update pi-apps (`~/pi-apps/updater` and press a key if it tells you) and try again.

**Comment by fabianmendes** [comment]
Amizing, that helped. Thanks!

**Comment by fabianmendes** [comment]
```
/home/pi/pi-apps/apps/Pycharm CE/install: l\u00ednea 55: 
/root/.local/share/applications/jetbrains-pycharm-ce.desktop: No existe el fichero o el directorio
```
_(Directory doesn't exist)_
![2021-03-19-184752_800x480_scrot](https://user-images.githubusercontent.com/31230743/111849324-b3c64000-88e3-11eb-855a-b02e73e19ef6.png)

**Comment by Itai-Nelken** [comment]
@techcoder20

**Comment by techcoder20** [comment]
Did you run pi-apps with `sudo` cause the script specifies this path `~/.local/share/applications/jetbrains-pycharm-ce.desktop
`

**Comment by CleanMachine1** [comment]
@techcoder2, I'm pretty sure @fabianmendes did due to the folder being /root/.local/.........

Try running just `pi-apps`

**Comment by Botspot** [comment]
I think all these issues were caused by running Pi-Apps with `sudo`. A complete reinstall may be necessary to avoid future permission errors.

**Comment by fabianmendes** [comment]
How should I reinstall pi-apps software easily?
 ~V. Fabian T· Mendes C.

**Comment by Botspot** [comment]
> How should I reinstall pi-apps software easily?

Where do you currently have it installed? Is it located at `/home/pi/pi-apps`?

**Comment by fabianmendes** [comment]
Yep, yes it is!

On Mar 24, 2021 1:58 PM, "Botspot" ***@***.***> wrote:

How should I reinstall pi-apps software easily?

Where do you currently have it installed? Is it located at /home/pi/pi-apps?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Botspot/pi-apps/issues/480#issuecomment-806038791>, or
unsubscribe

**Comment by fabianmendes** [comment]
I did that but got this then:
![2021-03-25-002914_800x480_scrot](https://user-images.githubusercontent.com/31230743/112419123-347ca600-8d01-11eb-8318-444ae517a9bc.png)
```

Installing Pycharm CE with install script
Running pkg-install...
Waiting until APT locks are released... Done
Failed to run sudo apt update!
APT reported these errors:
W: Failed to fetch https://download.puffinbrowser.com/repo/dists/stable/InRelease  Temporary failure resolving 'download.puffinbrowser.com'
W: Some index files failed to download. They have been ignored, or old ones used instead.


```

**Comment by CleanMachine1** [comment]
Where you running apt at the time or using something like automatic updates because apt can only do one thing at a time, and it creates a lock file to stop anything else happening

**Comment by fabianmendes** [comment]
No wherelse, I mean terminal is open because of pi-apps pycharm
installation, no oher terminal window was open. Also, I could install other
apps (Emoji font, for example)

**Comment by CleanMachine1** [comment]
Did you install 2 apps near the same time?

**Comment by fabianmendes** [comment]
No, man

**Comment by fabianmendes** [comment]
See on the screenshot above there is just one terminal window open

**Comment by fabianmendes** [comment]
I don't get it :L 😭😭

**Comment by techcoder20** [comment]
Try reinstalling puffin browser,  looks like there is something wrong with its repo

**Comment by CleanMachine1** [comment]
I have 2 theories:
Its trying to update puffin and its not making a connection

There is a break in one of the install scripts


Could you please open a terminal and run the command 'sudo apt update'

**Comment by CleanMachine1** [comment]
@fabianmendes ?

**Comment by Botspot** [comment]
> I did that but got this then:
> ```
> Failed to run sudo apt update!
> APT reported these errors:
> W: Failed to fetch https://download.puffinbrowser.com/repo/dists/stable/InRelease  Temporary failure resolving 'download.puffinbrowser.com'
> W: Some index files failed to download. They have been ignored, or old ones used instead.
> ```

This isn't a problem with Pycharm, but with your `apt` system.

You can run `sudo apt update` separately and see that it still fails. My guess is that the puffin browser developers messed up, or maybe your system installed it incorrectly.

I'd first try reinstalling puffin browser.

**Comment by fabianmendes** [comment]
Sorry I was traveling. Looking into it, let me do that sudo apt up rate for
you

**Comment by fabianmendes** [comment]
> I have 2 theories:
> Its trying to update puffin and its not making a connection
> 
> There is a break in one of the install scripts
> 
> Could you please open a terminal and run the command 'sudo apt update' and copy its output and post it here

@Itai-Nelken 
```

Err:1 https://itai-nelken.github.io/weekly-box86-debs/debian  InRelease        
  Fallo temporal al resolver \u00abitai-nelken.github.io\u00bb
Err:2 http://raspbian.raspberrypi.org/raspbian buster InRelease                
  Fallo temporal al resolver \u00abraspbian.raspberrypi.org\u00bb
Err:3 http://archive.raspberrypi.org/debian buster InRelease                   
  Fallo temporal al resolver \u00abarchive.raspberrypi.org\u00bb
Err:4 https://download.puffinbrowser.com/repo stable InRelease                 
  Fallo temporal al resolver \u00abdownload.puffinbrowser.com\u00bb
Leyendo lista de paquetes... Hecho          
Creando \u00e1rbol de dependencias       
Leyendo la informaci\u00f3n de estado... Hecho
Todos los paquetes est\u00e1n actualizados.
W: Fallo al obtener http://raspbian.raspberrypi.org/raspbian/dists/buster/InRelease  Fallo temporal al resolver \u00abraspbian.raspberrypi.org\u00bb
W: Fallo al obtener https://itai-nelken.github.io/weekly-box86-debs/debian/InRelease  Fallo temporal al resolver \u00abitai-nelken.github.io\u00bb
W: Fallo al obtener https://download.puffinbrowser.com/repo/dists/stable/InRelease  Fallo temporal al resolver \u00abdownload.puffinbrowser.com\u00bb
W: Fallo al obtener http://archive.raspberrypi.org/debian/dists/buster/InRelease  Fallo temporal al resolver \u00abarchive.raspberrypi.org\u00bb
W: No se han podido descargar algunos archivos de \u00edndice, se han omitido, o se han utilizado unos antiguos en su lugar.

```

**Comment by CleanMachine1** [comment]
@fabianmendes are you even connected to internet?

also can you please ping one of the severs with
`ping raspbian.raspberrypi.org
`

**Comment by Itai-Nelken** [comment]
@fabianmendes as @CleanMachine1 said it seems `apt` is having trouble connecting to the internet.

**Comment by fabianmendes** [comment]
OKok let me try again, thanks!

On Mar 27, 2021 5:11 PM, "Itai Nelken" ***@***.***> wrote:

@fabianmendes <https://github.com/fabianmendes> as @CleanMachine1
<https://github.com/CleanMachine1> said it seems apt is having trouble
connecting to the internet.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Botspot/pi-apps/issues/480#issuecomment-808802976>, or
unsubscribe

**Comment by fabianmendes** [comment]
Finally, I could install it... but, mhm, uh:
...
![2021-03-29-233620_800x480_scrot](https://user-images.githubusercontent.com/31230743/112943781-ed772200-90ff-11eb-9f0a-1a1179b3ebf5.png)

**Comment by techcoder20** [comment]
Run the installation for pycharm again, it should update the java version.( Command in script: `"${DIRECTORY}/pkg-install" "python-pip openjdk-11-jdk" "$(dirname "$0")" || exit 1`) If that does not work you can manually change java version by installing it manually with apt first (`sudo apt -y install openjdk-11-jdk`)and then changing the default version by following this link https://www.xmodulo.com/change-default-java-version-linux.html :)

**Comment by fabianmendes** [comment]
Alright I will

On Mar 30, 2021 5:04 AM, "RPICoder" ***@***.***> wrote:

Run the installation for pycharm again, it should update the java version.(
Command in script: "${DIRECTORY}/pkg-install" "python-pip openjdk-11-jdk"
"$(dirname "$0")" || exit 1) If that does not work you can manually change
java version by installing it manually with apt first (sudo apt -y install
openjdk-11-jdk)and then changing the default version by following this link
https://www.xmodulo.com/change-default-java-version-linux.html :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Botspot/pi-apps/issues/480#issuecomment-810050336>, or
unsubscribe

**Comment by fabianmendes** [comment]
> Run the installation for pycharm again, it should update the java version.( Command in script: `"${DIRECTORY}/pkg-install" "python-pip openjdk-11-jdk" "$(dirname "$0")" || exit 1`) If that does not work you can manually change java version by installing it manually with apt first (`sudo apt -y install openjdk-11-jdk`)and then changing the default version by following this link https://www.xmodulo.com/change-default-java-version-linux.html :)

What openjdk do you have? this always happen to me on 62-bit pycharm version on my 32-bit RPi OS version... How can I install newers jdk > 11 version? I saw the steps you've shared but I didn't get it.

**Comment by fabianmendes** [comment]
Which should I select? 
![2021-03-30-111445_800x480_scrot](https://user-images.githubusercontent.com/31230743/113012891-2e475900-9149-11eb-9088-13127fe2f7a3.png)

**Comment by techcoder20** [comment]
Try `2`

**Comment by fabianmendes** [comment]
AMAZING. Thaanks! @techcoder20

**Comment by fabianmendes** [comment]
Hi, any help with numpy?

I got this message, seems something being missing? Otherwise, I have installed it with `pip` and pip3` on my system as well.

```
/home/pi/PycharmProjects/trading/venv/bin/python /home/pi/PycharmProjects/trading/trying.py
Traceback (most recent call last):
  File "/home/pi/PycharmProjects/trading/venv/lib/python3.7/site-packages/numpy/core/__init__.py", line 22, in <module>
    from . import multiarray
  File "/home/pi/PycharmProjects/trading/venv/lib/python3.7/site-packages/numpy/core/multiarray.py", line 12, in <module>
    from . import overrides
  File "/home/pi/PycharmProjects/trading/venv/lib/python3.7/site-packages/numpy/core/overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: libf77blas.so.3: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pi/PycharmProjects/trading/trying.py", line 1, in <module>
    import numpy as np
  File "/home/pi/PycharmProjects/trading/venv/lib/python3.7/site-packages/numpy/__init__.py", line 145, in <module>
    from . import core
  File "/home/pi/PycharmProjects/trading/venv/lib/python3.7/site-packages/numpy/core/__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.7 from "/home/pi/PycharmProjects/trading/venv/bin/python"
  * The NumPy version is: "1.20.2"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: libf77blas.so.3: cannot open shared object file: No such file or directory


Process finished with exit code 1

```
![2021-04-23-163656_800x480_scrot](https://user-images.githubusercontent.com/31230743/115927983-59c11900-a453-11eb-93be-06dcbd993d99.png)
![2021-04-23-164011_800x480_scrot](https://user-images.githubusercontent.com/31230743/115927962-4f068400-a453-11eb-87ae-33716d2e1baf.png)
![2021-04-23-164231_800x480_scrot](https://user-images.githubusercontent.com/31230743/115927971-53cb3800-a453-11eb-9d29-8552b2ced573.png)
**This is the code quoted.  ^**
I have  more screenshots if you need.

**Comment by CleanMachine1** [comment]
Try running : pip3 install numpy

**Comment by fabianmendes** [comment]
> Try running : pip3 install numpy

I did it, it's already installed.

**Comment by CleanMachine1** [comment]
Thats all the suggestions I have sorry!

**Comment by fabianmendes** [comment]
Jajajaja j;

**Comment by CleanMachine1** [comment]
I have 1 more suggestion, is Pycharm using python3.7?

As running "python" in rpi os gives a python2.7 shell 

In your photo type the long string into a terminal and see if it opens a python3 or 2 shell

If its 2 we can make a symbolic link to python3.7

**Comment by techcoder20** [comment]
Python 3 is installed by default but the default version of python is set to 2. You need to change it manually. And I dont think its a good idea to mess with that as sometimes it can break apps.
